### PR TITLE
adding some more keys for the controller

### PR DESCRIPTION
**Pull request by roshanch**
This adds some more controller-keys from the oculus.

### CODE REVIEW

**Comment by roshanch** [comment]
Oculus doesn't seem to implement GVRArmModel interface, right now all other methods are not implemented in OVRControllerReader.

File: GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrControllerReader.java [hunk 1]
```diff
@@ -26,7 +26,62 @@
...
+    public float getPointerTiltAngle(){
+        return 0f;
+    }
+    @Override
+    public boolean getUseAccelerometer(){
+        return false;
+    }
+    @Override
+    public float getFadeDistanceFromFace(){
+        return 0f
```
Comment (dnvasilev) [inline GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrControllerReader.java]: Missing semicolon. One wrong key press in the wrong window at the wrong time (after compile, before commit)

**Comment by liaxim** [comment]
Doesn't build for me:
```
/home/mihail/git/gvrfmaster/GearVRf/GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGamepadDeviceManager.java:106: error: GVRGamepadController is not abstract and does not override abstract method setGazeBehavior(GazeBehavior) in GVRArmModel
    private static class GVRGamepadController extends GVRBaseController {
                   ^
/home/mihail/git/gvrfmaster/GearVRf/GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java:86: error: GVRMouseController is not abstract and does not override abstract method setGazeBehavior(GazeBehavior) in GVRArmModel
    private static class GVRMouseController extends GVRBaseController {
                   ^
/home/mihail/git/gvrfmaster/GearVRf/GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRGazeCursorController.java:32: error: GVRGazeCursorController is not abstract and does not override abstract method setGazeBehavior(GazeBehavior) in GVRArmModel
final class GVRGazeCursorController extends GVRBaseController implements GVRDrawFrameListener {
      ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 errors
```

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java [hunk 2]
```diff
@@ -25,9 +25,12 @@
 import org.gearvrf.io.CursorControllerListener;
 import org.gearvrf.io.GVRControllerType;
 import org.gearvrf.io.GVRInputManager;
+import org.gearvrf.utility.Log;
 import org.joml.Quaternionf;
 import org.joml.Vector3f;
 
+import static android.webkit.ConsoleMessage.MessageLevel.LOG;
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java]: Please remove this unnecessary import

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java [hunk 3]
```diff
@@ -46,17 +49,34 @@
  * notification whenever the controller information is updated.
  */
 final class GearCursorController extends GVRCursorController {
-    private static final int OVR_BUTTON_A = 0x00000001;
-    private static final int OVR_BUTTON_ENTER = 0x00100000;
-    private static final int OVR_BUTTON_BACK = 0x00200000;
 
+    public enum OVR_KEYS{
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java]: Second that. Please use more generic names.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java [hunk 4]
```diff
@@ -46,17 +49,34 @@
...
      * Defines the handedness of the gear controller.
      */
     public enum Handedness {
         LEFT, RIGHT
     }
-
+    /**
+     * Defines the GazeBehavior of the gear controller.
+     */
+    public enum GazeBehavior {
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java]: What exactly is this gaze behavior signifying?

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRArmModel.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  * It is responsible for:
+  * Determining the orientation and location of the controller.
+  * Predict the location of the shoulder, elbow, wrist, and pointer.
+  *
+  */
+
+package org.gearvrf;
+import org.joml.Vector3f;
+
+public interface GVRArmModel{
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRArmModel.java]: We can't expose an ARM model that doesn't work for all major backends. Please remove or make backend-specific.

Comment (roshanch) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRArmModel.java]: talked with Tom, I am removing armModel for now. I will make a separate pull request for other backends.

**Comment by dnvasilev** [comment]
Needs rebase

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java [hunk 6]
```diff
@@ -424,7 +447,7 @@ else if(thread.prevButtonEnter == KeyEvent.ACTION_UP && touching){
 
     private int handleAButtton(int key){
```
Comment (dnvasilev) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java]: Count number of t in handleAButtton

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java [hunk 7]
```diff
@@ -424,7 +447,7 @@ else if(thread.prevButtonEnter == KeyEvent.ACTION_UP && touching){
...
         long time = SystemClock.uptimeMillis();
-        int handled = handleButton(key, OVR_BUTTON_A, thread.prevButtonA, KeyEvent.KEYCODE_A);
+        int handled = handleButton(key, CONTROLLER_KEYS.BUTTON_A.getNumVal(), thread.prevButtonA, KeyEvent.KEYCODE_A);
```
Comment (dnvasilev) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GearCursorController.java]: Second parameter of handleButton should take enum type

**Comment by liaxim** [comment]
Looks good to me. Planning to merge on Thursday.

**Pull request merged**